### DISCUSSION TITLE
Get Teams, which have access to a repository

### DISCUSSION THREAD

**Discussion by fabge** [community/community#48324] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I would like to query the **collaborators and teams** who have access to a GitHub repo via GraphQL API.
I have an example from the UI of what return values I would expect to see (`fabian.dev-admin`):

![image](https://user-images.githubusercontent.com/21140791/220767375-3a5bf872-afb4-4e88-94af-8fbf6919a960.png)

The following GraphQL query returns the collaborators:

```
query {
  repository(owner: "EnBWAG-Test", name: "action.fabian.whatthe") {
    collaborators(first: 10) {
      nodes {
        login
      }
    }
  }
}
```

But I wasn't able (or it isn't possible?) to query the teams. The `repository` type in GraphQL doesn't have a `teams` field, `collaborators` only returns users.
Somehow, however, GitHub has to make the query itself to display this in the UI - and I thought that the GraphQL API might also be used in the backend - so there should be a way?

**Comment by martinm82** [comment]
Is there any solution on this problem? I am as well trying to obtain teams assigned to a repository but haven't found a solution yet

**Comment by byrneh** [comment, marked as answer]
@fabge I believe the answer is there is no GraphQL API to directly get the teams aspects of your requirement.
You can however use the REST API "[List repository teams](https://docs.github.com/en/rest/repos/repos?list-repository-teams=&apiVersion=2022-11-28#)" GET /repos/{owner}/{repo}/teams for the teams aspect of your requirement

**Reply by atsu85** [reply]
GraphQl query for this would be really useful (to avoid n+1 queries), as i have hundreds of repos :(

**Reply by byrneh** [reply]
@atsu85 that is a slightly different ask/question than the original thread post.
If your hundreds of repo use case related to a small number of organizations it is possible to
Query Organization edge node Teams (a list of items on the organization) 
and the edge node repositories within teams (a list of repositories this team has access to) there's also a query string on repositories if you wish to perform filtering based on repository name 
The repository edge for the team has a field "permission" (the permission level the team has on the repository).

In effect this would produce a report of all teams in an orgnanization and permissions each team has on any repositories.
